src/pykeen/optimizers/rsgd.py

```diff
@@ -20,10 +22,17 @@
 class RiemannianSGD(Optimizer):
     """A variant of :class:`torch.optim.SGD` generalized for riemannian manifolds."""
 
-    def __init__(self, params, lr=required, param_names=None):
+    def __init__(self, params: Sequence[Parameter], param_names: Sequence[str], poincare: Set[str], lr=required):
```

Comment:
iirc, `params` and `param_names` should match in length?

Then why not provide `params: Mapping[str, Parameter]` instead?